src/main/kotlin/de/itemis/mps/gradle/runmigrations/Plugin.kt

```diff
@@ -0,0 +1,83 @@
+    val force: Property<Boolean> = of.property(Boolean::class.java).convention(false)
+}
+@Suppress("unused")
+open class RunMigrationsMpsProjectPlugin : Plugin<Project> {
+    override fun apply(project: Project) {
+        project.run {
+            val extension = extensions.create("runMigrations", MigrationExecutorPluginExtensions::class.java)
+            afterEvaluate {
+                val mpsLocation = extension.mpsLocation.convention{ File(project.buildDir, "mps") }.map { it.asFile }.get()
+                val projectLocation = extension.projectLocation.getOrElse { throw GradleException("No project path set") }
+                if(!file(projectLocation).exists()) {
+                    throw GradleException("The path to the project doesn't exist:$projectLocation")
+                }
+                val forceMigration = extension.force
+                
+                val mpsVersion = extension.getMPSVersion()
+                val parsedMPSVersion = SemVer.parse(mpsVersion)
+                if(forceMigration.getOrElse(false) && (parsedMPSVersion.major < 2021 || (parsedMPSVersion.major == 2021 && parsedMPSVersion.minor < 3))) {
+                    throw GradleException("The force migration flag is only supported for 2021.3.0 and higher.")
+                }
+                
+                val resolveMps: Task = if (extension.mpsConfig.isPresent) {
+                    tasks.create("resolveMpsForMigrations", Copy::class.java) {
+                        from({extension.mpsConfig.get().resolve().map { zipTree(it) }})
+                        into(mpsLocation)
+                    }
+                } else {
+                    tasks.create("resolveMpsForMigrations")
+                }
+                
+                tasks.register("runMigrations") {
+                    dependsOn(resolveMps)
+                    doLast {
+                        ant.withGroovyBuilder { 
+                            "path"("id" to "path.mps.ant.path",) {
+                                // The different MPS versions need different jars. Let's just keep it simple and include all jars.
+                                "fileset"("dir" to  "$mpsLocation/lib", "includes" to "**/*.jar")
+                            }
+                            "taskdef"("resource" to "jetbrains/mps/build/ant/antlib.xml", "classpathref" to "path.mps.ant.path")
+                        }
+                        if(forceMigration.getOrElse(false)) {
+                            ant.withGroovyBuilder {
+                                "migrate"("project" to projectLocation, "mpsHome" to mpsLocation, "force" to true) {
+                                    "macro"("name" to "mps_home", "path" to mpsLocation)
+                                    "jvmargs"() {
+                                        "arg"("value" to "-Didea.log.config.file=log.xml")
```

Comment:
I think we should make heap size configurable, i.e. let the user specify a value and we convert it to a "-Xmx" argument.